src/python/txtai/cloud/storage.py

```diff
@@ -45,15 +45,39 @@ def __init__(self, config):
         # Get driver for provider
         driver = get_driver(config["provider"])
 
-        # Get client connection
-        self.client = driver(
-            config.get("key", os.environ.get("ACCESS_KEY")),
-            config.get("secret", os.environ.get("ACCESS_SECRET")),
-            host=config.get("host"),
-            port=config.get("port"),
-            token=config.get("token"),
-            region=config.get("region"),
-        )
+        if config.get("token") and config.get("region"):
```

Comment:
I'm thinking we can do this in a simpler fashion to reduce the duplicate code. 

Could something like this work?

```python
        # Get client connection
        self.client = driver(
            config.get("key", os.environ.get("ACCESS_KEY")),
            config.get("secret", os.environ.get("ACCESS_SECRET")),
            **{field: config.get(field) for field in ["host", "port", "token", "region"] if config.get(field)}
        )
```